src/components/connectionError/connectionError.jsx

```diff
@@ -0,0 +1,53 @@
+import React, { useEffect, useState } from 'react';
+import {
+  View, Text, TouchableOpacity,
+} from 'react-native';
+import NetInfo from '@react-native-community/netinfo';
+import PropTypes from 'prop-types';
+
+import i18n from '../../i18n';
+import styles from './connectionError.styles';
+
+function ConnectionError({ isClosable }) {
+  const [errorText, setErrorText] = useState('');
+  useEffect(() => {
+    const unsubscribe = NetInfo.addEventListener(({ isConnected }) => {
+      if (!isConnected) {
+        setErrorText('no connection');
+      } else {
+        setErrorText('');
+      }
+    });
+    return unsubscribe;
+  }, []);
+  if (errorText) {
+    return (
+      <View style={styles.connectionErrorView}>
+        <Text style={styles.connectionErrorText}>
+          {i18n.t(errorText)}
+        </Text>
+        {isClosable && (
+          <TouchableOpacity
+            style={styles.closeButton}
+            onPress={() => setErrorText('')}
+          >
+            <Text style={styles.closeButtonText}>
+              ×
+            </Text>
+          </TouchableOpacity>
+        )}
+      </View>
+    );
+  }
+  return null;
+}
+
+ConnectionError.propTypes = {
+  isClosable: PropTypes.bool,
+};
+
+ConnectionError.defaultProps = {
+  isClosable: true,
+};
+
+export default ConnectionError;
```

Comment:
Don't forget to use `React.memo` for functional components. (doesn't matter in this case)